### DISCUSSION TITLE
Wazuh dashboard keeps reloading after implementing sigma rules

### DISCUSSION THREAD

**Discussion by mrkasali** [wazuh/wazuh#15976] [Q&A]
Hey there!
So after implementing sigma rules (about 3700) my dashboard keeps reloading every 30 seconds or so.
But it only reloads when using the security event dashboard or mitre framework.
When using the events tab everything works just fine.

![wazuh](https://user-images.githubusercontent.com/88803929/213441063-c472fde2-1b21-4baf-8bf1-2b56f864ed7d.PNG)

Anyone has an idea why this happens?
Cpu and ram usage are normal.

**Comment by mdiego92** [comment]
Hello @mrkasali,

Could you please share with us the output of the following command?

`cat /var/ossec/logs/ossec.log | grep -iE 'ERR|WARN|CRIT'
`
Thank you very much!

**Reply by mrkasali** [reply]
Heres the filtered file!
Sadly theres a lot of mitre errors because some IDs are missing in the database.
Had to zip it because its to big (55MB)
[ossec.zip](https://github.com/wazuh/wazuh/files/10456738/ossec.zip)

**Reply by mdiego92** [reply]
@mrkasali, if the rules had any syntax error you shouldn't be able to start the Wazuh Manager. If you remove the Sigma rules, does the application work properly?

Also, I saw a few log entries pointing to the permissions of the /var/ossec/etc/shared folder and subfolder files.
Could you please confirm if the ownership of these files is `wazuh:wazuh`?

Thank you!

**Reply by mrkasali** [reply]
Yes i just tested it again. After moving the sigma.xml out of the rules folder everything works fine.

Thanks for pointing it out!
Indeed most of the files were root:root.
Changed it to wazuh:wazuh.

**Comment by kmfreder1** [comment]
When we first implemented sigma rules, we found that many of them contained ```<if_group>windows</if_group>``` which cause an explosion of rules in the rule set, because every rule with that setting ends up hanging off of each and every windows rule.  This caused all kinds of strange behavior in Wazuh and in logtest.  You might also check for a circular reference in the new ruleset, like in a group tagged as sysmon, is there any ```<if_group>sysmon</if_group>``` in any of the rules.  What source did you use for your sigma rules?  We were using https://github.com/SanWieb/sigWah, unfortunately these haven't been updated in 2 years.  Hope this helps.

**Reply by mrkasali** [reply]
I just checked the rules and theres no `<if_group>windows</if_group>` or `<if_group>sysmon</if_group>` at all.
I used this converter: https://github.com/theflakes/sigma_to_wazuh and the original sigma rules from https://github.com/SigmaHQ/sigma
2656 out of the 3709 rules use this: `<if_sid>18100, 60000, 60001, 60002, 60003, 60004, 60006, 60007, 60008, 60009, 60010, 60011, 60012</if_sid>`

**Comment by mrkasali** [comment]
Just tried it again but this time i got an error while wazuh was loading:

![wazuh](https://user-images.githubusercontent.com/88803929/213470326-2bfdc13f-e3d6-425a-b23b-a90d454e5a21.PNG)

Refreshed the page and it instantly worked again.

**Reply by mdiego92** [reply]
@mrkasali, did you change anything in your environment, or just waited and refreshed the web?

I'm glad it is now resolved. If you still want we can analyze the API logs, I would need to see the output of `cat /var/ossec/logs/api.log`?

Please let me know if it happens again! Thank you very much for the update on this.

**Reply by mrkasali** [reply]
@mdiego92, I didnt change anything before refreshing.
Sigma rules were still active and the wazuh dashboard kept refreshing afterwards.
But this was the first time i got this error and ever since the error didnt appear.
I can confirm that it has to do something with the sigma rules.
Right now i deleted the .xml file so i can check the alerts and i dont have any problems.
Only when i create the .xml for the sigma rules the problem occurs.
I will send you the api.log as soon as possible!
Thank you very much for the help!

**Reply by mdiego92** [reply]
Hello @mrkasali,

It would be great if you could reproduce the error and send me the /var/ossec/logs/api.log and /var/ossec/logs/ossec.log file with the log entry. Also, If it's possible I can check your sigma rules in case there is anything wrong with the syntax.

Please let me know. I'll gladly help you with this!

**Reply by mrkasali** [reply]
@mdiego92 I´m very sorry for the late answer but I was very busy the last few days.
So i was able to reproduce this error the 1st time i tried.
But i have to mention that theres an API version mismatch because there went something wrong when updating the mitre.db
The version mismatch is not the cause for this problem. The first time i got this error was before updating the mitre.db
Heres the api.log and ossec.log!
[ossec.log](https://github.com/wazuh/wazuh/files/10500395/ossec.log)
[api.log](https://github.com/wazuh/wazuh/files/10500351/api.log)

**Reply by mrkasali** [reply]
@mdiego92 Oh and here are the sigma rules generated with this tool: https://github.com/theflakes/sigma_to_wazuh
[sigma.txt](https://github.com/wazuh/wazuh/files/10500418/sigma.txt)

**Reply by mdiego92** [reply]
Hello @mrkasali,

- Could you please confirm what version of Wazuh are you using?
- Also, it would be great if you could share with me the specific rules so I can test them locally, as it seems they're generating conflicts with the manager: `2023/01/25 08:55:06 wazuh-analysisd: CRITICAL: (1220): Error loading the rules: 'etc/rules/local_rules.xml'.`

Thank you very much!

**Reply by mrkasali** [reply]
@mdiego92 the version is 4.3.10 right now.
The critical errors occured while testing arround.
At the moment there arent any critical errors!

**Reply by mdiego92** [reply]
Hello @mrkasali,

Just following-up. Is everything working properly?
Kind regards!
